.dev/get_urls.R

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/Rscript
+# Script to get URL keys from CRAN
+#   of packags that Suggest or Import lintr
+
+cran_db = as.data.frame(available.packages())
+lintr_re = "\\blintr\\b"
+lintr_pkg = cran_db[
+  with(cran_db, grepl(lintr_re, Suggests) | grepl(lintr_re, Imports)),
+  "Package"
+]
+
+# simple retry mechanism
+desc_url_fmt = "https://cran.r-project.org/web/packages/%s/DESCRIPTION"
+extract_url = function(pkg, initial_sleep = 1, retry_sleep = 60) {
+  desc_url = sprintf(desc_url_fmt, pkg)
+  retry = function(cond) {
+    message("Sleepy... 😴")
+    Sys.sleep(retry_sleep)
+    url(desc_url, open = "r")
+  }
+  Sys.sleep(initial_sleep)
+  url_conn = tryCatch(url(desc_url, open = "r"), error = retry, warning = retry)
+  pkg_data = read.dcf(url_conn, "URL")
+  drop(pkg_data)
+}
+
+extract_github_repo = function(urls) {
+  matches = gregexpr("https?://git(hub|lab).com/[a-zA-Z0-9_-]+/[a-zA-Z0-9._-]+", urls)
+  starts = sapply(matches, `[[`, 1L)
+  matched = starts > 0L
+
+  urls = urls[matched]
+  starts = starts[matched]
+  matches = matches[matched]
+
+  unname(substr(urls, starts, starts + sapply(matches, attr, "match.length") - 1L))
+}
+
+urls = character(length(lintr_pkg))
+# for loop makes debugging easier
+for (ii in seq_along(urls)) {
+  urls[ii] <- extract_url(lintr_pkg[ii])
+}
+
+urls = sort(extract_github_repo(urls[!is.na(urls)]))
```

Comment:
Where we could add manual URLs.
Would be nice to automatically get a list of packages with missing git repos.
I noticed `fst` and `fstcore` for example.
Those are at `https://github.com/fstpackage/fst/` and `https://github.com/fstpackage/fstcore` respectively.

Reply:
Could crawl `BugReports` as well, that would have found fst: `"https://github.com/fstpackage/fst/issues"`

Reply:
we could also back up to the cran org mirrors on GitHub, emailing the maintainers if they don't have a repo listed

Reply:
problem is we won't see their .lintr... we could default to running _all_ our linters for those repos

Reply:
Let's see how many packages need to be manually found after crawling BugReports. Appsilon packages, e.g. shiny.semantic also provide a github issue link.

I wouldn't loose sleep if a few downstream packages can only be tested on the default lintr config.

Reply:
for the bugreports one... probably easier to just put those in the extra- manual file